cmd/minishift/cmd/profile/profile_set.go

```diff
@@ -41,6 +42,10 @@ func runProfile(cmd *cobra.Command, args []string) {
 	validateArgs(args)
```

Comment:
As @hferentschik  suggested, we need to add the logic to `validateArgs(args)` in profile.go.

Reply:
moved the check to _validateArgs_.

Reply:
+1, now please add some tests. Especially everything around validateArgs is easily testable since it is purely dependent on the CLI input (no running VM needed, etc). See also https://github.com/minishift/minishift/blob/4b9ddbf40f525d63a86eac087b1406997b1a5ce9/cmd/minishift/cmd/addon/addon_apply_test.go

You can call runProfile directly and then assert that the system exits with a specific error message.

Reply:
Thanks @hferentschik for the example, will add tests now. and about the other pull requests the arch linux documentation one got merged yesterday, i have updated #1621 PTAL when you get time, will have to do some work on #1640

Reply:
agree with @hferentschik here... I see many issues on your name and several unfinished PRs. Try not to start new work, unless it is working/merged.

Especially as you can use the time in-between to look at some usage and general knowledge about OpenShift.